<a href="https://colab.research.google.com/github/ManjaxX/Temas2023/blob/main/Lineas_de_espera.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Una sala de espera tiene una capacidad para 3 personas. Las personas arriban al sistema de acuerdo a una tasa de 8 por hora con distribución Poisson y son atendidas por una recepcionista en 10 minutos con distribución exponencial. Si alguien llega y el sistema está lleno, se retira sin entrar.

Las probabilidades de un paso son:
\begin{bmatrix}
  1-8\Delta t & 8\Delta & 0 & 0 & 0 \\
  6\Delta & 1-14\Delta & 8\Delta & 0 & 0 \\
  0 & 6\Delta & 1-14\Delta & 8\Delta & 0 \\
  0 & 0 & 6\Delta & 1-14\Delta & 8\Delta \\
  0 & 0 & 0 & 6\Delta & 1-6\Delta
  \end{bmatrix}

La cual nos da el siguiente sistema de ecuaciones:
\begin{equation}
\begin{split}
P_0 & = (1-8\Delta t)P_0 &+ (8\Delta t)P_1 \\
P_1 & = (6\Delta t)P_0 &+ (1-14\Delta t)P_1 &+ (8\Delta)P_2 \\
P_2 & =  &+ (6\Delta t)P_1 &+ (1-14\Delta t)P_2 &+ (8\Delta)P_3 \\
P_3 & = &+ &+ (6\Delta t)P_2 &+ (1-14\Delta t)P_3 &+ (8\Delta)P_4 \\
P_4 & = &+ &+ &+ (6\Delta t)P_3 &+ (1-6\Delta t)P_4 \\
1 & = P_0 &+ P_1 &+ P_2 &+ P_3 &+ P_4
\end{split}
\end{equation}

De la primera ecuación: $P_0 = (1-8\Delta t)P_0 + (6\Delta t)P_1$ encontramos el valor de $P_1$ en términos de $P_0$:

$$P_0=P_0-8ΔtP_0+6ΔtP_1$$
$$8ΔtP_0=6ΔtP_1$$
$$P_1=\frac{8}{6}P_0$$
de manera análoga se encuentran $P_2, P_3, P_4$
$$P_2=\left(\frac{8}{6}\right)^2P_0$$
$$P_3=\left(\frac{8}{6}\right)^3P_0$$
$$P_4=\left(\frac{8}{6}\right)^4P_0$$

Las probabilidades de estado estacionario $P_i$ representan el comportamiento probabilístico de cada estado del sistema a largo plazo por lo que $\sum P_i=1$

De lo anterior podemos entonces resolver para $P_0$:
$$P_0+P_1+P_2+P_3+P_4=1$$
$$P_0+\left(\frac{8}{6}\right)P_0+\left(\frac{8}{6}\right)^2P_0+\left(\frac{8}{6}\right)^3P_0+\left(\frac{8}{6}\right)^4P_0=1$$

In [ ]:
P_0=(1 + 8/6 + (8/6)**2 + (8/6)**3 + (8/6)**4)**(-1)
P_1=P_0*(8/6)
P_2=P_0*(8/6)**2
P_3=P_0*(8/6)**3
P_4=P_0*(8/6)**4
print("P_0=", P_0)
print("P_1=", P_1)
print("P_2=", P_2)
print("P_3=", P_3)
print("P_4=", P_4)

P_0= 0.10371318822023048
P_1= 0.1382842509603073
P_2= 0.18437900128040974
P_3= 0.2458386683738796
P_4= 0.32778489116517284


Ahora se puede calcular la tasa de entrada promedio. Este es el valor ponderado de las tasas de entrada al sistema y representan el número promedio de clientes que ingresan al sistema. A su vez, esta variable es la tasa de salida del sistema.

$$\bar{\lambda}=\sum\lambda_nP_n$$

In [ ]:
L = 8*(P_0)+8*(P_1)+8*(P_2)+8*(P_3)+0*(P_4)
print(L)

5.377720870678617


Por lo tanto se puede esperar que entran al sistema 5.377 personas cada hora. Lo que es un valor menor a los datos del problema de 8 personas por hora, esto es porque, el 32% del tiempo la sala está llena y se pierden clientes.

Ahora podemos calcular la utilización del servicio. Esto es, el porcentaje de tiempo que la recepcionista se encuentra atendiendo clientes.
$$\rho=\frac{\bar{λ}}{μ}=0.8963$$

In [ ]:
R=(L)/(6)
print(R)

0.8962868117797695


In [ ]:
L = 0*P_0+1*P_1+2*P_2+3*P_3+4*P_4
print(L)
Lq = 1*P_2+2*P_3+3*P_4
print(Lq)

2.555697823303457
1.6594110115236875


Con esto sabemos que el 89.6% del tiempo el servidor se encuentra ocupado, tiene capacidad suficiente para atender a los 5.37 clientes por hora. Finalmente calculamos el promedio del número de clientes en la sala de espera incluyendo al que esta siendo atendido:
$$L=\sum nP_n=2.55$$
y el promedio de clientes sólamente en la sala de espera es
$$L=\sum_{n=1}^{4} (n-1)P_n=1.66$$